app/controllers/users_controller.rb

```diff
@@ -20,15 +34,21 @@ def update
     @user = current_user
     if @user.update(update_params)
       render :show
+      if params[:unconfirmed_email].present?
+        @user.regenerate_email_confirmation_token
+        UserMailer.email_confirmation(@user, @user.unconfirmed_email).deliver_now
+      end
     else
       render status: :bad_request, json: current_user.errors
     end
   end
 
   def create_session
     @user = User.find_by(email: params[:email])
-    if @user&.authenticate(params[:password])
-      render json: { token: @user.auth_token }
+    if @user&.authenticate(params[:password]) && @user&.activated
+      render status: :ok, json: { token: @user.auth_token }
+    elsif @user&.authenticate(params[:password])
+      render status: :bad_request, json: { error: 'Please activate your account by following the instructions in the account confirmation email you received to proceed.' }
     else
       head :bad_request
```

Comment:
Error message is a good idea, add one for this too ("Invalid username or password" or similar)